src/Microsoft.OpenApi.Readers/ReferenceServices/IOpenApiVersionService.cs

```diff
@@ -31,5 +34,13 @@ public interface IOpenApiReferenceService
         /// <param name="type">The type of the reference.</param>
         /// <returns>The <see cref="OpenApiReference"/> object or null.</returns>
         OpenApiReference ConvertToOpenApiReference(string reference, ReferenceType? type);
+
+
+        /// <summary>
+        /// Function that converts MapNodes into Tag objects in a version specific way
+        /// </summary>
+        Func<MapNode, OpenApiTag> TagLoader { get; }
+
+        OpenApiDocument LoadOpenApi(RootNode rootNode);
```

Comment:
Comments?

Reply:
:-) Didn't want to spend too much time adding all the useful comments for you folks to decide it was a terrible idea!  I will add them.